### PR TITLE
chore: enables docker ecosystem for dependabot

### PR DESCRIPTION
**Pull request by nilekhc**
Signed-off-by: Nilekh Chaudhari <1626598+nilekhc@users.noreply.github.com>

**Reason for Change**:

**Requirements**

- [x] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [ ] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -12,3 +12,9 @@ updates:
       interval: "weekly"
     commit-message:
       prefix: "chore"
+  - package-ecosystem: "docker"
+    directory: "/"
```
Comment (aramase) [inline .github/dependabot.yml:16]: This should only be enabled for charts in `manifest_staging`

**Review by aramase — CHANGES_REQUESTED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (jeffwidman) [reply .github/dependabot.yml]: Just curious, I noticed you have a dockerfile in `"/"`:
https://github.com/Azure/secrets-store-csi-driver-provider-azure/blob/ed68f1895677a32095a58c6fecaa66b27a8c972a/Dockerfile

Any reason to not also have Dependabot watch that?

File: .github/dependabot.yml [hunk 2]
```diff
@@ -12,3 +12,9 @@ updates:
       interval: "weekly"
     commit-message:
       prefix: "chore"
+  - package-ecosystem: "docker"
+    directory: "/manifest_staging"
```
Comment (jeffwidman) [inline .github/dependabot.yml:16]: The check unfortunately isn't recursive, so you'll need to manually specify the actual directory containing the chart... I filed https://github.com/dependabot/dependabot-core/issues/6047 for us to evaluate changing that behavior.

Comment (jeffwidman) [reply .github/dependabot.yml:16]: And looks like there's another bug, even when you _do_ get it configured correctly: https://github.com/dependabot/dependabot-core/issues/6048

I'll try to debug ☝️ tomorrow.

**Comment by jeffwidman** [comment]
I enabled the `kubernetes_updates` feature flag on this repo, so Dependabot will be trying to update Docker image tags within Helm/K8s files.

Once I get https://github.com/dependabot/dependabot-core/issues/6048 fixed, hopefully it will start working and opening PR's for you.

**Comment by jeffwidman** [comment]
Also, thanks again @nilekhc for offering this repo as a test case, even for using in my own fork it's been great to have much more realistic data for testing than anything I'd put together quickly.

File: .github/dependabot.yml (hunk 2 above)
Comment (nilekhc) [reply .github/dependabot.yml:16]: @jeffwidman Do you still want me to update the path and point it to the `values.yaml` directory?

**Comment by nilekhc** [comment]
> I enabled the `kubernetes_updates` feature flag on this repo, so Dependabot will be trying to update Docker image tags within Helm/K8s files.
> 
> Once I get [dependabot/dependabot-core#6048](https://github.com/dependabot/dependabot-core/issues/6048) fixed, hopefully it will start working and opening PR's for you.

@jeffwidman Do you still want me to update the path and point it to the `values.yaml` directory?

**Comment by nilekhc** [comment]
> Also, thanks again @nilekhc for offering this repo as a test case, even for using in my own fork it's been great to have much more realistic data for testing than anything I'd put together quickly.

We are happy to help! 🙂

**Comment by nilekhc** [comment]
> > I enabled the `kubernetes_updates` feature flag on this repo, so Dependabot will be trying to update Docker image tags within Helm/K8s files.
> > Once I get [dependabot/dependabot-core#6048](https://github.com/dependabot/dependabot-core/issues/6048) fixed, hopefully it will start working and opening PR's for you.
> 
> @jeffwidman Do you still want me to update the path and point it to the `values.yaml` directory?

@jeffwidman Any word on this?

**Comment by jeffwidman** [comment]
Yes, please do point the path at the `values.yaml` directories... we won't support recursion out of the gate, and not sure if/when we would do that as there's a lot more product questions/edge cases, so you'll need to manually specify any directories that you want watched.

Beyond that, I figured out why it wasn't working on my fork of your repo, and fixed some other bugs along the way:
* https://github.com/dependabot/dependabot-core/pull/6094
* https://github.com/dependabot/dependabot-core/pull/6095
* https://github.com/dependabot/dependabot-core/pull/6096
* https://github.com/dependabot/dependabot-core/pull/6097

Once the last one is merged/deployed, this should all start working... but we have an internal deploy freeze because of GitHub Universe so I won't be able to deploy til late tomorrow or Friday.

**Review by jeffwidman — APPROVED** [review]

**Comment by jeffwidman** [comment]
All the PR's have been merged, and this now works correctly on my fork... once you merge, can you confirm it works correctly here as well? You should get a PR opened by Dependabot similar to the ones here: https://github.com/jeffwidman/secrets-store-csi-driver-provider-azure/pulls

**Review by aramase — APPROVED** [review]
lgtm

**Pull request merged**